addons/mail/static/src/messaging/entity/activity/activity.js

```diff
@@ -30,8 +30,34 @@ function ActivityFactory({ Entity }) {
             this.delete();
         }
 
+        /**
+         * Opens (legacy) form view dialog to edit current activity and update
+         * it when dialog is closed.
+         */
+        async edit() {
+            const action = {
+                type: 'ir.actions.act_window',
+                name: this.env._t("Schedule Activity"),
+                res_model: 'mail.activity',
+                view_mode: 'form',
+                views: [[false, 'form']],
+                target: 'new',
+                context: {
+                    default_res_id: this.res_id,
+                    default_res_model: this.res_model,
+                },
+                res_id: this.id,
+            };
+            return this.env.do_action(action, {
```

Comment:
use `await` or document the return value, or even use none of those if we don't actually care when it ends (same debate we had over the other PR about action/on_close, where I think we decided to not wait).